### ISSUE TITLE
Scramble user apiKey in database

### ISSUE THREAD

**Issue by ll4strw** [rspace-os/rspace-web#58]
UserApiKey.apiKey entries in db should be possibly scrambled.

**Comment by mKowalski256** [comment]
Yes that's a fair request, although in current design we do allow user to view/copy their API key [anytime after its creation](https://documentation.researchspace.com/article/v0dxtfvj7u-rspace-api-introduction#obtaining_your_api_key), which means most secure scrambling mechanism based on hashes would not work in this case. We could encrypt/decrypt the keys based on a secret stored outside database, but that would only be a partial improvement.

We encourage users to revoke/regenerate their API keys on the schedule they believe is reasonable.

**Comment by mKowalski256** [comment]
> Privacy. sysadmins would have access to all API keys and impersonate users outside the impersonations scheme provided by the GUI app itself

Hmm, I've checked that and I believe you mean that 'operate as' functionality currently let sysadmin see the API key of the user? That's a bug/oversight, the API key shouldn't be displayed for sysadmin, I'll raise it as a separate issue.

**Comment by mKowalski256** [comment]
I see that outside of 'operate as' scenario there is an API endpoint allowing sysadmin to retrieve API keys. The issue https://github.com/rspace-os/rspace-web/issues/73 will cover both cases.

**Comment by mKowalski256** [comment]
Hi Leonardo, after internal discussion we've prioritized this ticket, and we're targeting the fix for 2.3/1.103 release (end of September). We plan to go with solution where user see the API key only once, at the moment of creation, after which RSpace stores just a hashed value of the key.

**Comment by rs-nicof** [comment]
I can confirm this has been handled by the JIRA ticket https://researchspace.atlassian.net/browse/RSDEV-310 and will be released in the RSpace version `1.103`.

@mKowalski256 not sure if this github issue needs to be closed now

**Comment by mKowalski256** [comment]
This change is merged and part of 2.3.0 open-source release (and in 1.103.0 supported release) - API keys are now stored hashed with SHA256, and only visible to the user immediately after creation.

**Comment by mKowalski256** [comment]
For reference, the change was implemented with [commit 57d9b](https://github.com/rspace-os/rspace-web/commit/57d9b99984e92f7435e81e08a565ce9b1ce907ee). 

There is a new deployment property `sysadmin.apikey.generation`, set to `false` by default, which allows System Admin to revoke/regenerate API keys when 'operating as' the user.
